### PR TITLE
Fix for windows;

### PR DESCRIPTION
**Pull request by vigneshpa**
Executing deno from pure JS

### CODE REVIEW

**Comment by vigneshpa** [comment]
#14

**Review by kt3k — COMMENTED** [review]
Thanks for the PR. Left one question

**Comment by vigneshpa** [comment]
> Thanks for the PR. Left one question

Sorry didn't notice that.
The deno binary should already be there if the npm module is installed properly.
I saw that you are invoking the install.js when the binary file is not found. So I worte the 10th line to do the same.

File: bin/deno.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env node
+const fs = require("fs");
+const chile_process = require("child_process");
+const path = require("path");
+
+const executableFilename = process.platform==="win32"?"deno.exe":"deno";
+
+const executablePath = path.join(__dirname, executableFilename);
+
+if(!fs.existsSync(executablePath)) require("./install");
```
[the remark this answers is not in the document]
Comment (vigneshpa) [reply bin/deno.js:10]: Removing the 10th should be fine if the npm module is installed properly.

[the remark this answers is not in the document]
Comment (vigneshpa) [reply bin/deno.js:10]: We could ask the user to reinstall the npm module if the deno binary is not found

[the remark this answers is not in the document]
Comment (kt3k) [reply bin/deno.js:10]: This module installs `deno` executable on `install` script https://github.com/kt3k/deno-bin/blob/d80a9f603be6eb53dc5feeb3bd3235a3cfe07b4d/package.json#L15

But the users can disable `install` script explicitly by `--ignore-scripts` option. The purpose of this script is to support those cases. (See also https://github.com/kt3k/deno-bin/pull/12

So asking user reinstalling doesn't make sense here.

I think what we need to do here is:
- make `main` function in `install.js` returning a promise
- export `main` function from `install.js`
- require `main` function here
- await `main` function call

[the remark this answers is not in the document]
Comment (vigneshpa) [reply bin/deno.js:10]: That's a good idea

[the remark this answers is not in the document]
Comment (vigneshpa) [reply bin/deno.js]: How is it now

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Comment by vigneshpa** [comment]
Thank you for your response.
Kindly do merge.

**Pull request merged**